### PR TITLE
SWIFT-1117 Add a README

### PR DESCRIPTION
**Pull request by kmahar**
README for the repo. Let me know if you think there's anything else we should cover here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,145 @@
...
+            .product(name: "MongoDBVapor", package: "mongodb-vapor")
+        ])
+    ]
+)
+```
+
+Then run `swift build` to download, compile, and link all your dependencies.
+
+## Example Usage
+### You can find a complete example project built using this library [here](https://github.com/mongodb/mongo-swift-driver/tree/main/Examples/VaporExample).
```
Comment (kmahar) [inline README.md:74]: this is intentionally duplicative of top section to increase the chance people see it 🙂

File: README.md [hunk 2]
```diff
@@ -1,2 +1,145 @@
...
+```swift
+// Configure the app for using a MongoDB server at the provided connection string.
+try app.mongoDB.configure("mongodb://localhost:27017")
+let coll = app.mongoDB.client.db("home").collection("kittens")
+// creates a unique index if it doesn't exist already.
+_ = try coll.createIndex(["name": 1], indexOptions: IndexOptions(unique: true)).wait()
+```
+
+### Working with Extended JSON
+If you'd like to use `ExtendedJSONEncoder` and `ExtendedJSONDecoder` for encoding/decoding JSON requests/responses, in  `App/configure.swift`, add:
```
Comment (patrickfreed) [inline README.md:140]: I think it's probably worth pointing out that if you want to use a `BSONDocument` directly as `Content`, you'll need to do this configuration. Once we implement regular `Codable` conformance we can remove the warning.

Edit: also may want to link to the JSON guide to provide more information. We also probably want to update that JSON guide to include this information as well.

**Review by patrickfreed — COMMENTED** [review]
Looks good, just one suggestion

File: README.md (hunk 2 above)
Comment (kmahar) [reply README.md:140]: added; making follow-up PR to BSON repo to update JSON guide.

Comment (kmahar) [reply README.md:140]: actually looks like we already have all of that there - Nellie had added a section which I updated a couple weeks ago that has basically the same content and notes not all BSON types work with JSONEncoder/JSONDecoder

https://github.com/mongodb/swift-bson/blob/main/Guides/JSON-Interop.md#using-extended-json-with-vapor

**Review by patrickfreed — APPROVED** [review]

**Pull request merged**